classes/submission/SubmissionDAO.inc.php

```diff
@@ -452,6 +465,114 @@ function getByStatus($status, $stageUserId = null, $reviewUserId, $contextId = n
 	}
 
 	/**
+	 * Get all submissions that are considered assigned to the passed user.
+	 * @param $userId int
+	 * @param $contextId int optional
+	 * @param $title string|null optional Filter by title.
+	 * @param $author string|null optional Filter by author.
+	 * @param $stageId int|null optional Filter by stage id.
+	 * @param $rangeInfo DBResultRange optional
+	 * @return DAOResultFactory
+	 */
+	function getAssignedToUser($userId, $contextId = null, $title = null, $author = null, $stageId = null, $rangeInfo = null) {
+		$params = $this->_getFetchParameters();
+		$userId = (int) $userId;
+		array_push($params, $userId, STATUS_DECLINED, $userId, $userId, ROLE_ID_AUTHOR, $userId);
+		if ($contextId) $params[] = $contextId;
+
+		if ($title) {
+			$params[] = AppLocale::getLocale();
+			$params[] = 'title';
+			$params[] = '%' . $title . '%';
+		}
+		if ($author) array_push($params, $authorQuery = '%' . $author . '%', $authorQuery, $authorQuery);
+		if ($stageId) $params[] = $stageId;
+
+		$result = $this->retrieveRange($sql =
+			'SELECT s.*, ps.date_published,
+				' . $this->_getFetchColumns() . '
+			FROM submissions s
+				LEFT JOIN published_submissions ps ON (s.submission_id = ps.submission_id)
+				LEFT JOIN stage_assignments sa ON (s.submission_id = sa.submission_id)
+				LEFT JOIN submission_files sf ON (s.submission_id = sf.submission_id)
+				LEFT JOIN signoffs so ON (sf.file_id = so.assoc_id) 
+				LEFT JOIN user_groups g ON (g.user_group_id = so.user_group_id)
+				LEFT JOIN review_assignments ra ON (s.submission_id = ra.submission_id)
+				' . ($title?' LEFT JOIN submission_settings ss ON (s.submission_id = ss.submission_id)':'') . '
+				' . ($author?' LEFT JOIN authors au ON (s.submission_id = au.submission_id)':'')
```

Comment:
What if there are two authors for the same submission whose names both match? That'll cause it to be listed twice, I think.